langchain4j-ollama/src/main/java/dev/langchain4j/model/ollama/OllamaChatModel.java

```diff
@@ -82,26 +96,18 @@ public static OllamaChatModelBuilder builder() {
     public Response<AiMessage> generate(List<ChatMessage> messages) {
         ensureNotEmpty(messages, "messages");
 
-        ChatRequest request = ChatRequest.builder()
-                .model(modelName)
-                .messages(toOllamaMessages(messages))
-                .options(options)
-                .format(format)
-                .stream(false)
-                .build();
-
-        ChatResponse response = withRetry(() -> client.chat(request), maxRetries);
-
-        return Response.from(
-                AiMessage.from(response.getMessage().getContent()),
-                new TokenUsage(response.getPromptEvalCount(), response.getEvalCount())
-        );
+        return doGenerate(messages, null);
     }
 
     @Override
     public Response<AiMessage> generate(List<ChatMessage> messages, List<ToolSpecification> toolSpecifications) {
         ensureNotEmpty(messages, "messages");
+        ensureNotEmpty(toolSpecifications, "toolSpecifications");
```

Comment:
not sure about this change, it might break for someone

Reply:
OK. I revert this check now.